### PR TITLE
recipe-images/opentrons-image-ot3: Create Bootable Tezi image + Opentrons update build

### PR DESCRIPTION
**Pull request by vegano1**
[RCORE-173](https://opentrons.atlassian.net/browse/RCORE-173)

Overview:
These changes produce a bootable Tezi image for flashing as well as the updatable image used for the opentrons app. Note the rootfs is mounted as read-write for now, so we can move along with Opentrons updates.

Changes:
- Remove custom image_type_tezi_ot3.bbclass and integrate tezi image generation into our own opentrons-ot3-image recipe
- Separate the rootfs into a systemfs for system applications (robot-server, update-server, etc) and systemfs for user level changes.
- Create a modified image.json file for the Tezi image that creates two partitions for our A/B update mechanism and one partition for the userfs.
- Use the generated Tezi image fragments to build our own Tezi tarball with our generated systemfs and userfs
- Create a VERSION.json file to be used for updates via the Opentrons app
- Create the ot3-system.zip with VERSION.json systemfs.ext4, systemfs 256 hash
- Create a recipe (userfs-mount.bb) that installs systemd mount units for mounting the /userfs as well as bind-mounting /var, /home, and /data to it.
- Remove wks file

### CODE REVIEW

File: recipes-images/images/opentrons-ot3-image.bb [hunk 1]
```diff
@@ -53,10 +37,178 @@ IMAGE_INSTALL += " \
...
+# create the tezi image.json
+python do_create_tezi_manifest(){
+    import os
+    import json
+    tezi_manifest_path = "%s/image-%s.json" % (d.getVar('DEPLOY_DIR_IMAGE'), d.getVar('IMAGE_BASENAME'))
+    tezi_ot3_manifest_path = "%s/image.json" % (d.getVar('DEPLOY_DIR_IMAGE'))
+    tezi_manifest = {}
+
+    # define the ot3  partitions
+    ot3_partitions = [{
```
Comment (sfoster1) [inline recipes-images/images/opentrons-ot3-image.bb:93]: This doesn't quite look right. We want
- boot, FATFS, 48 meg
- RFS1, holding systemfs, fixed size (here, 3072M; would be good to cut this down if we can)
- RFS2, holding systemfs, same size as RFS1
- VARFS or whatever, holding userfs, remaining space on the storage

Right now, we've got bootfs and RFS1, but the volume labeled REFS2 has the userfs, and there's no dedicated partition for it.

Having two dedicated rootfs partitions allows us to update the partition we're not booted from.

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Looks almost great, we need to fix up the partition allocation and then we're solid I think.

We're not currently mounting the userfs, right?

**Review by sfoster1 — APPROVED** [review]
This looks really good to me now! I still need to build an image and mount and poke around in some root filesystems to be sure but I think this is great.

**Review by sfoster1 — APPROVED** [review]
Excellent, looks good to me!

My only quibble is that 2 gigs is pretty darn big for the rootfs, it'll make updates very slow - even if the filesystems compress well, they still have to be written. Do you have an idea of how much space is actually used in them?

**Comment by vegano1** [comment]
> Excellent, looks good to me!
> 
> My only quibble is that 2 gigs is pretty darn big for the rootfs, it'll make updates very slow - even if the filesystems compress well, they still have to be written. Do you have an idea of how much space is actually used in them?

Currently, we are using ~900mb, I've reduced the rootfs partition size from 2GB to 1.5GB. 
We can adjust as needed, don't want to bring it too low so we don't run into the same issue as OT2. 
That being said we will be doing some trimming in the near future and can adjust as needed.

**Pull request merged**